string direction
float32 duration
float32 speed
